client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -24,8 +25,16 @@ const newsletter: Flow = {
 		return translate( 'Newsletter' );
 	},
 	useSteps() {
+		const query = useQuery();
+		const isComingFromMarketingPage = query.get( 'ref' ) === 'newsletter-lp';
+
 		return [
-			{ slug: 'intro', asyncComponent: () => import( './internals/steps-repository/intro' ) },
+			// Load intro step component only when not coming from the marketing page
+			...( ! isComingFromMarketingPage
+				? [
+						{ slug: 'intro', asyncComponent: () => import( './internals/steps-repository/intro' ) },
+				  ]
+				: [] ),
```

Comment:
I found the optional spread be simpler to look at, but it's also "one liner trickery". Happy to revert back to previous if you think that looked better. :-)

Reply:
works for me